lib/internal/modules/esm/resolve.js

```diff
@@ -1087,6 +1063,40 @@ function defaultResolve(specifier, context = {}) {
     // Ignore exception
   }
 
+  // Import maps are processed first so that policies can still apply
+  // to the resolved specifier and also so the other resolution logic
+  // happens on the mapped specifier
+  if (importMap) {
+    // Intentionally mutating here as we don't think it is a problem
+    // since it is a primitive
+    specifier = importMap.resolve(parsed || specifier, parsedParentURL);
+  }
+
+  if (parentURL && policy?.manifest) {
+    const redirects = policy.manifest.getDependencyMapper(parentURL);
+    if (redirects) {
+      const { resolve, reaction } = redirects;
+      const destination = resolve(specifier, new SafeSet(conditions));
+      let missing = true;
+      if (destination === true) {
+        missing = false;
+      } else if (destination) {
+        const href = destination.href;
+        return { __proto__: null, url: href };
+      }
+      if (missing) {
+        // Prevent network requests from firing if resolution would be banned.
+        // Network requests can extract data by doing things like putting
+        // secrets in query params
+        reaction(new ERR_MANIFEST_DEPENDENCY_MISSING(
+          parentURL,
+          specifier,
+          ArrayPrototypeJoin([...conditions], ', ')),
+        );
+      }
+    }
+  }
```

Comment:
If you don't mind, let's extract this into its own function (so we can unit test it in a follow-up).

Reply:
Yeah good idea. I was just not changing things originally which were not necessary, but I think this approach makes sense since the lines will be blamed to me either way.